insights/tests/datasources/test_leapp.py

```diff
@@ -1,51 +1,67 @@
 import json
 import pytest
 
-from mock.mock import Mock
+try:
+    from unittest.mock import patch
+    builtin_open = "builtins.open"
+except Exception:
+    from mock import patch
+    builtin_open = "__builtin__.open"
+
 from os import path
 
-from insights.core.exceptions import SkipComponent
-from insights.specs.datasources.leapp import leapp_report, LocalSpecs
+from insights.core.exceptions import SkipComponent, ContentException
+from insights.specs.datasources.leapp import leapp_report
 
 
 with open(path.join(path.dirname(__file__), 'leapp-report.json'), 'r') as fp:
-    INPUT_OK = fp.readlines()
+    INPUT_OK = json.load(fp)
 
 with open(path.join(path.dirname(__file__), 'leapp-report.ret'), 'r') as fp:
     RESULT = fp.readlines()
 
-INPUT_NG_1 = """
+INPUT_NG_1 = json.loads("""
 { "entries":[
   {
-    "a": "B",
+    "a": "B"
   }
  ]
 }
-""".strip()
-
-INPUT_NG_2 = ""
+""".strip())
 
+INPUT_NG_2 = json.loads("{}")
 
-def test_leapp_report():
-    report_file = Mock()
-    report_file.content = INPUT_OK
-    broker = {LocalSpecs.leapp_report_raw: report_file}
-
-    result = leapp_report(broker)
 
+@patch("os.path.isfile", return_value=True)
+@patch(builtin_open)
+@patch("json.load", return_value=INPUT_OK)
+def test_leapp_report_ok(mopen, misfile, mload):
```

Comment:
Hi @xiangce , 

I am all good with the code, except one concern here: 

I know these args won't be referenced here. 
Still, it would be better to align the order of these args' (based on the meaning) with the `@patch` order.  
Here is a print example of the args during test running: 
```
<MagicMock name='load' id='4609748944'> <MagicMock name='open' id='4609831120'> <MagicMock name='isfile' id='4609856720'>
```
Alternatively, replace these args with some meaningless arg names will also do.

Reply:
Good catch, done!